noir-projects/aztec-nr/aztec/src/oracle/notes.nr

```diff
@@ -221,7 +228,22 @@ unconstrained fn get_app_tagging_secret_oracle(
     _recipient: AztecAddress,
 ) -> [Field; INDEXED_TAGGING_SECRET_LENGTH] {}
 
-pub unconstrained fn increment_app_tagging_secret(sender: AztecAddress, recipient: AztecAddress) {
+/// Notifies the simulator that a tag has been used in a note, and to therefore increment the associated index so that
+/// future notes get a different tag and can be discovered by the recipient.
+/// This change should only be persisted in a non-volatile database if the tagged log is found in an actual block -
```

Comment:
Yes, we'd keep the local index. My comment there is purposely a bit naive to not delve too much on the intrincacies of how PXE actually works (it will likely not have an index cache).